src/wasm/wasm-ir-builder.cpp

```diff
@@ -649,11 +668,14 @@ Result<> IRBuilder::visitTryStart(Try* tryy, Name label) {
 }
 
 Result<> IRBuilder::visitTryTableStart(TryTable* trytable, Name label) {
+  applyDebugLoc(trytable);
```

Comment:
Why do we `applyDebugLoc` in control flow structures in this file but not to e.g. `visitStructSet`?

Reply:
For normal instructions, the `applyDebugLoc` in `push` is sufficient. For control flow, however, we need to apply the debug location at the beginning of the scope. Since the control flow isn't `push`ed until the end of the scope, we can't rely on the same `applyDebugLoc` as for normal instructions.

Reply:
Ah, yes, thanks, that makes sense, and I think we have similar code for the old parse in fact...